### PR TITLE
Fix Reassign/ShiftEdit updating role requirements and leaves badly

### LINKED ISSUE #195: Editing shift removes leaves
Using `shift-edit` to change the roles from a blank shift that has no role requirements result in leaves being removed from that shift.

### PR DESCRIPTION
**Pull request by WangZijun97**
Resolves #195 
Ensures that Reassign/ShiftEdit updates both old and new shift's role requirements. Resolves that issue where roles seemingly cannot be added if though there should be available slots.

### CODE REVIEW

**Review by wakululuu — APPROVED** [review]
LGTM

File: src/main/java/mcscheduler/logic/commands/ShiftEditCommand.java [hunk 1]
```diff
@@ -116,7 +117,8 @@ private void editShiftInAssignments(Model model, Shift shiftToEdit, Shift edited
         for (Assignment assignment : fullAssignmentList) {
             if (shiftToEdit.isSameShift(assignment.getShift())) {
                 Role assignmentRole = assignment.getRole();
-                if (!newRoles.contains(assignmentRole) || assignment.getWorker().isUnavailable(editedShift)) {
+                if ((!newRoles.contains(assignmentRole) && !assignmentRole.equals(new Leave()))
```
Comment (wakululuu) [inline src/main/java/mcscheduler/logic/commands/ShiftEditCommand.java]:
```suggestion
                if ((!newRoles.contains(assignmentRole) && !Leave.isLeave(assignmentRole))
```

File: src/main/java/mcscheduler/model/shift/Shift.java [hunk 2]
```diff
@@ -86,6 +86,11 @@ public boolean isRoleRequired(Role role) {
      */
     public static void updateRoleRequirements(Model model, Shift shiftToUpdate, Role role) {
         CollectionUtil.requireAllNonNull(model, shiftToUpdate, role);
+        assert model.hasShift(shiftToUpdate);
+        if (role.equals(new Leave())) {
```
Comment (wakululuu) [inline src/main/java/mcscheduler/model/shift/Shift.java]:
```suggestion
        if (Leave.isLeave(role)) {
```

**Review by wakululuu — COMMENTED** [review]
Just a minor suggestion to use `Leave#isLeave` instead of `Leave#equals`

**Comment by WangZijun97** [comment]
Thanks for suggestions, added as requested.

**Pull request merged**